api/hypershift/v1beta1/hostedcluster_types.go

```diff
@@ -1857,8 +1857,98 @@ type AzurePlatformSpec struct {
 	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="SecurityGroupID is immutable"
 	// +kubebuilder:validation:Required
 	// +immutable
-	// +required
 	SecurityGroupID string `json:"securityGroupID,omitempty"`
+
+	// managedIdentities contains the client IDs related to the managed identities needed for HCP control plane
+	// and data plane components that authenticate with Azure's API.
+	//
+	// +kubebuilder:validation:Required
+	ManagedIdentities AzureResourceManagedIdentities `json:"managedIdentities,omitempty"`
+}
+
+// AzureResourceManagedIdentities contains the client IDs related to the managed identities needed for HCP control plane
+// and data plane components that authenticate with Azure's API.
+type AzureResourceManagedIdentities struct {
+	// ControlPlane contains the client IDs of all the managed identities on the HCP control plane needing to
+	// authenticate with Azure's API.
+	//
+	// +kubebuilder:validation:Required
+	ControlPlane ControlPlaneManagedIdentities `json:"controlPlane"`
+
+	// Future placeholder - DataPlaneMIs * DataPlaneManagedIdentities
+}
+
+// ManagedIdentityClientID is a client ID of a managed identity
+// +kubebuilder:validation:XValidation:rule="self.matches('^[{]?[0-9a-fA-F]{8}-([0-9a-fA-F]{4}-){3}[0-9a-fA-F]{12}[}]?$')",message="the client ID of a managed identity must be a valid UUID. It should be 5 groups of hyphen separated hexadecimal characters in the form 8-4-4-4-12."
+type ManagedIdentityClientID string
+
+// ControlPlaneManagedIdentities contains the client IDs of all the managed identities on the HCP control plane needing
+// to authenticate with Azure's API.
+// Managed identity regex pattern is from Microsoft here - https://learn.microsoft.com/en-us/azure/azure-resource-manager/management/resource-name-rules#microsoftmanagedidentity.
+// The format a managed identity should be `/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ManagedIdentity/userAssignedIdentities/{managedIdentityName}`.
+type ControlPlaneManagedIdentities struct {
+	// cloudProviderClientID is the client ID of a pre-existing managed identity ID associated with the azure
+	// cloud provider, aka ccm. The client ID of a managed identity must be a valid UUID. It should be 5 groups of
```

Comment:
I would avoid the abbrevation CCM and instead just say `Azure Cloud Controller Manager and Azure Cloud Node Manager` assuming this does also apply to the CNM.

Since this is part of the control plane group of client IDs, I guess not for the CNM, and that's separately authenticated?

Reply:
Ahh yes fair point, you don't have control plane nodes! Ok, all good